File: mindsdb-search-app/backend/utils/searchNews.py
export async function searchNews(query: string) {
  const response = await fetch("http://127.0.0.1:5000/search", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({ query }),
  });

  if (!response.ok) {
    const errData = await response.json();
    throw new Error(errData.error || "Something went wrong");
  }

  const data = await response.json();

  // Safely parse the format from ChromaDB
  const documents = data?.documents?.[0] || [];
  const metadatas = data?.metadatas?.[0] || [];

  return documents.map((content: string, i: number) => ({
    title: metadatas[i]?.title || `Result ${i + 1}`,
    content,
    source: metadatas[i]?.source || "Unknown",
  }));
}

File: components/searchNews.ts
export async function searchNews(query: string) {
  const apiUrl = process.env.NEXT_PUBLIC_API_URL || "http://127.0.0.1:8000";

  const response = await fetch(`${apiUrl}/api/search?q=${encodeURIComponent(query)}`, {
    method: "GET",
    headers: {
      "Content-Type": "application/json",
    },
  });

  if (!response.ok) {
    let errMessage = "Something went wrong";
    try {
      const errData = await response.json();
      errMessage = errData?.error || errMessage;
    } catch {
      errMessage = `HTTP ${response.status}: ${response.statusText}`;
    }
    throw new Error(errMessage);
  }

  const data = await response.json();

  console.log("✅ Backend data:", data); // Debug log (optional)

  // Normalize and return in frontend-friendly format
  return (data.results || []).map((item: any, i: number) => ({
    title: item.title || `Startup ${i + 1}`,
    content: item.content || "No description provided.",
    source: item.category || "Uncategorized",
  }));
}
